pkg/plan/lookup_query.go

```diff
@@ -36,3 +36,28 @@ func NewReverseLookupQuery(resourceID, kind string) *models.SearchRequest {
 		},
 	}}
 }
+
+// LookupByResourceIdQuery can be used to find a deployment by a resource-id (can be any kind e.g. elasticsearch, kibana, etc.)
+// (Builds a query that searches all possible kinds for the resource-id)
+func LookupByResourceIdQuery(resourceID string) *models.SearchRequest {
+	kinds := []string{"elasticsearch", "kibana", "apm", "appsearch", "enterprise_search", "integrations_server"}
```

Comment:
Nit: is there a way we can derive these rather than hardcode to future proof?

Reply:
There is no list of kinds in the API definition from what I can see, so we can't automatically get it.